src/main/java/com/autotune/common/datasource/prometheus/PrometheusDataOperatorImpl.java

```diff
@@ -127,18 +134,21 @@ public Object getValueForQuery(String url, String query) {
 
     /**
      * executes specified query on datasource and returns the JSON Object
-     * @param url String containing the url for the datasource
-     * @param query String containing the query to be executed
+     *
+     * @param url                   String containing the url for the datasource
+     * @param query                 String containing the query to be executed
+     * @param authenticationConfig
      * @return JSONObject for the specified query
      */
     @Override
-    public JSONObject getJsonObjectForQuery(String url, String query) {
-        GenericRestApiClient apiClient = new GenericRestApiClient(
-                CommonUtils.getBaseDataSourceUrl(
-                        url,
-                        KruizeConstants.SupportedDatasources.PROMETHEUS
-                )
-        );
+    public JSONObject getJsonObjectForQuery(String url, String query, AuthenticationConfig authenticationConfig) {
```

Comment:
notice that many functions are using authenticationConfig. This configuration should be set at the experiment level, similar to how experiment.setDatasource works. When retrieving the datasource using experiment.getDatasource, it should also return the associated authenticationConfig.

To achieve this, the authenticationConfig object should be linked to the datasource and accessible through it.